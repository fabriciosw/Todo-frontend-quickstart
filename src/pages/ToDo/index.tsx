import React, { useEffect, useState } from 'react';
import { Tarefa } from '../../contexts/TaskContext';
import { ITarefa } from '../../interfaces/tarefa';
import Formulario from '../../components/Form';
import * as todoServices from '../../services/todoServices';
import Lista from '../../components/List';
import { logoutUser } from '../../services/userServices';
// import { AuthenticationContext } from "../../contexts/AuthenticationContext";

const ToDo = (): JSX.Element => {
  const [taskStatus, setTaskStatus] = useState<boolean | undefined>(undefined);
  const [tarefas, setTarefas] = useState<ITarefa[]>([]);
  // const { setIsAuthenticated } = AuthenticationContext();

  useEffect(() => {
    todoServices.getAll(setTarefas);
  }, []);

  function verFinalizadas(): void {
    setTaskStatus(true);
  }
  function verPendentes(): void {
    setTaskStatus(false);
  }
  function verTodas(): void {
    setTaskStatus(undefined);
  }
  return (
    <Tarefa.Provider value={{ tarefas, setTarefas }}>
      <div className="App">
        <div className="leftside">
          <div className="formbox">
            <Formulario />

            <div className="linha">
              <button type="button" onClick={() => verTodas()}>
                Ver Todas
              </button>
              <button type="button" onClick={() => verPendentes()}>
                Ver Pendentes
              </button>
              <button type="button" onClick={() => verFinalizadas()}>
                Ver Conclu√≠das
              </button>
            </div>
            <button
              type="button"
              onClick={
                () => logoutUser()
                // setIsAuthenticated
              }
            >
              Deslogar
            </button>
          </div>
        </div>
        <div className="rightside">
          <h1 className="title">Tarefas</h1>
          <Lista taskStatus={taskStatus} />
        </div>
      </div>
    </Tarefa.Provider>
  );
};

export default ToDo;
