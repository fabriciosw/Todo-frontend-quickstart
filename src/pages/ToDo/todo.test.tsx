import React from 'react';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import ToDo from '.';

describe('Main screen', () => {
  it('should match snapshot', () => {
    const { container } = render(<ToDo />);

    expect(container).toMatchSnapshot();
  });
});
