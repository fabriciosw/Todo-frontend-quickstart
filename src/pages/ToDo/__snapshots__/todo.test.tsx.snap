// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main screen should match snapshot 1`] = `
<div>
  <div
    class="App"
  >
    <div
      class="leftside"
    >
      <div
        class="formbox"
      >
        <h1>
          Lista de tarefas
        </h1>
        <div
          class="linha"
        >
          <h5>
            total: 
            0
          </h5>
          <h5>
            pendentes: 
            0
          </h5>
          <h5>
            concluídas: 
            0
          </h5>
        </div>
        <form>
          <div
            class="campos"
          >
            <label
              for="tarefa"
            >
              Adicione um a nova tarefa
            </label>
            <input
              id="tarefa"
              name="tarefa"
              placeholder="Nome da tarefa"
              required=""
              type="text"
              value=""
            />
            <label
              for="descricao"
            >
              Adicione uma descrição
            </label>
            <input
              id="descricao"
              name="descricao"
              placeholder="Descricao da tarefa"
              type="text"
              value=""
            />
            <input
              type="submit"
            />
          </div>
        </form>
        <div
          class="linha"
        >
          <button>
            Ver Todas
          </button>
          <button>
            Ver Pendentes
          </button>
          <button>
            Ver Concluídas
          </button>
        </div>
      </div>
      <button>
        Deslogar
      </button>
    </div>
    <div
      class="rightside"
    >
      <h1
        class="title"
      >
        Tarefas
      </h1>
      <aside />
    </div>
  </div>
</div>
`;
